### PR TITLE
feat(notificationdrawer): add recommendation option

### LINKED ISSUE #3312: Notification drawer to support teal variation
It seems that the notification drawer does not currently support the teal notification variation and it would be great if that could be added.

We have added a 'Recommendation' section to the notification drawer in the OpenShift console and are using the teal variation to identify notifications in that particular grouping. 

Note: we are currently using an existing variant name "update"

@christiemolloy 
@mceledonia 

Notifications-1

### PR DESCRIPTION
**Pull request by christiemolloy**
closes #3312

### CODE REVIEW

File: src/patternfly/components/NotificationDrawer/examples/NotificationDrawer.md [hunk 1]
```diff
@@ -139,6 +139,7 @@ cssPrefix: pf-c-notification-drawer
 | `.pf-m-warning` | `.pf-c-notification-drawer__list-item` | Modifies a notification list item for the warning state. |
 | `.pf-m-danger` | `.pf-c-notification-drawer__list-item` | Modifies a notification list item for the danger state. |
 | `.pf-m-success` | `.pf-c-notification-drawer__list-item` | Modifies a notification list item for the success state. |
+| `.pf-m-recommendation` | `.pf-c-notification-drawer__list-item` | Modifies a notification list item for the recommendation state. |
```
Comment (mcoker) [inline src/patternfly/components/NotificationDrawer/examples/NotificationDrawer.md:142]: Just want to verify this should be called `pf-m-recommendation` over `pf-m-default`?

File: src/patternfly/components/NotificationDrawer/notification-drawer-basic-list.hbs [hunk 2]
```diff
@@ -33,6 +33,23 @@
...
   {{/notification-drawer-list-item}}
+  {{#> notification-drawer-list-item notification-drawer-list-item--IsRecommendation="true"}}
+    {{#> notification-drawer-list-item-header}}
+      {{> notification-drawer-list-item-header-icon}}
+      {{#> notification-drawer-list-item-header-title}}
+        Unread recommendation notification title. This is a long title to show how the title will wrap if it is long and wraps to multiple lines.
+      {{/notification-drawer-list-item-header-title}}
+    {{/notification-drawer-list-item-header}}
+    {{#> notification-drawer-list-item-action}}
+      {{#> dropdown id=(concat notification-drawer--id "-action2") dropdown-menu--modifier="pf-m-align-right" dropdown--IsActionMenu="true" dropdown-toggle--modifier="pf-m-plain" dropdown--HasKebabIcon="true" aria-label="Actions"}}{{/dropdown}}
```
Comment (mcoker) [inline src/patternfly/components/NotificationDrawer/notification-drawer-basic-list.hbs:44]: duplicate ID `notification-drawer--id "-action2"`

**Comment by mcoker** [comment]
Looks good! Just a couple of things.

File: src/patternfly/components/NotificationDrawer/examples/NotificationDrawer.md (hunk 1 above)
Comment (christiemolloy) [reply src/patternfly/components/NotificationDrawer/examples/NotificationDrawer.md:142]: @megan-hall @mceledonia @mcarrano what are your thoughts here?

Comment (mcarrano) [reply src/patternfly/components/NotificationDrawer/examples/NotificationDrawer.md:142]: Thanks for flagging this @mcoker . Even though 'recommendation' conforms to the OpenShift use case, we don't use that term elsewhere in PatternFly. We defined the teal alert state as 'default' that can be used for anything that the consumer wants. Can we call it 'pf-m-default'?

Comment (christiemolloy) [reply src/patternfly/components/NotificationDrawer/examples/NotificationDrawer.md]: updated

**Comment by christiemolloy** [comment]
Updated @mcoker @mcarrano

**Pull request merged**